### PR TITLE
[feature] add classes which map the tables

### PR DESCRIPTION
**Pull request by barteked**
## 1. Before

The application contained only one entity class representing the simplest database table. The remaining tables that do not include any foreign key constraints were not yet mapped in the code.

## 2. Problem

The lack of entity classes and repositories for these tables made it impossible to interact with their data using Spring Data. Additionally, the `initial_data.sql` file did not include any test data, so the database was empty after starting the Docker container, which made it impossible to test data reading.

## 3. Solution

I created `@Entity` classes for all simple tables (those without foreign key relations). For each of them, I also created the corresponding `JpaRepository` interface. I added one INSERT statement per table to the initial_data.sql file and started a new Docker container with the database. I verified that reading all rows from these tables works as expected.

## 4. Future Steps
	•	Create entity classes and repositories for tables containing foreign key relations

### CODE REVIEW

File: src/main/java/pl/kedrabartosz/HomeBudget/version2/entities/ItemEntity.java [hunk 1]
```diff
@@ -0,0 +1,22 @@
...
+@ToString
+@Table(name = "item")
+@Entity
+public class ItemEntity {
+    @Id
+    @Column(name = "id")
+    private Integer id;
+    @Column(name = "name")
+    private String name;
+    @Column(name = "category_id")
```
Comment (PaniOdProgramowania) [inline src/main/java/pl/kedrabartosz/HomeBudget/version2/entities/ItemEntity.java:18]: We wanted just the `easy` classes first which are not connected to other tables. Like `Category` for example. 

We'll do the linking between tables is topic on the next lesson.

Therefore please apply the following changes in this and other files:
- delete all fields which are links between tables, that is which are foreign keys
- delete all `@ToString` annotations in `Entity` classes
- Please also update the PR's description to follow this pattern:
```
1. Before - here you'll describe the current state of code, like "there were no classes representing tables from the db"
2. Problem - here you'll state the problem, why was it that we wanted the before-state to change, why was it problematic 
3. Solution - here you'll describe what you did (for example "I created entity classes for simple tables without foreign keys")
4. Future Steps - here you'll write what are your next steps (like in step 2 - I won't give you an example, can you think of what are the future steps here?)
```

**Review by PaniOdProgramowania — CHANGES_REQUESTED** [review]
Please adjust the code to fulfill requirements.

File: src/main/java/pl/kedrabartosz/HomeBudget/HomeBudgetApplication.java [hunk 2]
```diff
@@ -6,18 +6,47 @@
 import org.springframework.boot.SpringApplication;
 import org.springframework.boot.autoconfigure.SpringBootApplication;
 import org.springframework.context.ConfigurableApplicationContext;
-import pl.kedrabartosz.HomeBudget.repository.ItemRepository;
-import pl.kedrabartosz.HomeBudget.repository.ItemRepositoryFactory;
-import pl.kedrabartosz.HomeBudget.repository.FileBasedCategoryRepository;
-import pl.kedrabartosz.HomeBudget.repository.ListBasedItemRepository;
-import pl.kedrabartosz.HomeBudget.service.ItemService;
-import pl.kedrabartosz.HomeBudget.service.ReceiptService;
+import pl.kedrabartosz.HomeBudget.version1.Item;
```
Comment (PaniOdProgramowania) [inline src/main/java/pl/kedrabartosz/HomeBudget/HomeBudgetApplication.java:9]: wrong imports. We mostly want only `version2` now. If there are names clashes, please use the long imports for `version1`. We want `version2` to be the new standard

File: src/main/java/pl/kedrabartosz/HomeBudget/version2/entities/CategoryEntity.java [hunk 3]
```diff
@@ -0,0 +1,22 @@
+package pl.kedrabartosz.HomeBudget.version2.entities;
+
+import jakarta.persistence.Column;
+import jakarta.persistence.Entity;
+import jakarta.persistence.Id;
+import jakarta.persistence.Table;
+
+
+import java.time.Instant;
+@Table(name = "category")
```
Comment (PaniOdProgramowania) [inline src/main/java/pl/kedrabartosz/HomeBudget/version2/entities/CategoryEntity.java:10]: please add new lines between `imports` section and `class` section :)

**Review by PaniOdProgramowania — COMMENTED** [review]
Added some open questions and 1 comment. See video about the description in the chat

File: src/main/java/pl/kedrabartosz/HomeBudget/HomeBudgetApplication.java (hunk 2 above)
Comment (barteked) [reply src/main/java/pl/kedrabartosz/HomeBudget/HomeBudgetApplication.java:9]: done

File: src/main/java/pl/kedrabartosz/HomeBudget/version2/entities/ItemsInReceiptEntity.java [hunk 4]
```diff
@@ -0,0 +1,16 @@
...
+import jakarta.persistence.Table;
+
+@Table(name = "items_in_receipt")
+@Entity
+public class ItemsInReceiptEntity {
+    @Id
+    @Column(name = "id")
+    private String id;
+    @Column(name = "quantity")
+    private Integer quantity;
```
[the remark this answers is not in the document]
Comment (barteked) [reply src/main/java/pl/kedrabartosz/HomeBudget/version2/entities/ItemsInReceiptEntity.java:15]: I changed the type from Integer to int because the field in the data justification has a NOT NULL constraint, so nullability does not apply.

**Review by PaniOdProgramowania — APPROVED** [review]
🎉
